content/actions/reference/workflow-syntax-for-github-actions.md

```diff
@@ -688,7 +688,17 @@ steps:
 
 #### Custom shell
 
-You can set the `shell` value to a template string using `command […options] {0} [..more_options]`. {% data variables.product.prodname_dotcom %} interprets the first whitespace-delimited word of the string as the command, and inserts the file name for the temporary script at `{0}`.
+You can set the `shell` value to a template string using `command […options] {0} [..more_options]`. {% data variables.product.prodname_dotcom %} interprets the first whitespace-delimited word of the string as the command, and inserts the file name for the temporary script at `{0}`, as in this example:
+
+```yaml
+steps:
+  - name: Display the environment variables and their values
+    run: |
+      print %ENV
+    shell: perl {0}
+```
+
+The command used, `perl` in this case, needs to be installed in the environment, please check [the documentation on virtual environments](https://github.com/actions/virtual-environments) for the list of commands, languages and utilities available.
```

Comment:
I think this alternative link will give the user the info they need while still staying on the docs site (which can lead to a smoother user experience). I also made some small stylistic changes.

```suggestion
The command used, `perl` in this example, must be installed on the runner. For information about the software included on GitHub-hosted runners, see "[Specifications for GitHub-hosted runners](/actions/reference/specifications-for-github-hosted-runners#supported-software)."
```